### ISSUE TITLE
Radii

### ISSUE THREAD

**Issue by jameskoster** [WordPress/gutenberg#63703]
<img width="1852" alt="radius" src="https://github.com/WordPress/gutenberg/assets/846565/285a29b1-d525-40b0-9867-78acdf47acd8">

Recent work has expanded the values used for border radius in the UI. Specifically; the new menu component, document title, and command palette all make use of a larger 4px radius. The preview / content frame, and modal radius is larger still: 8px.

A defined scale for radius can add polish to the UI. For example compare how menu items nest neatly inside their container in the new menu component, compared with the current one:

| DropdownMenu V2 | DropdownMenu |
| --- | --- |
| <img width="233" alt="Screenshot 2024-07-11 at 16 32 17" src="https://github.com/WordPress/gutenberg/assets/846565/6eae501a-45fb-4ce5-a26e-ba0315570ef3"> | <img width="221" alt="Screenshot 2024-07-11 at 16 32 26" src="https://github.com/WordPress/gutenberg/assets/846565/4500cdb7-3dc7-494c-9847-62ce89f56eb2"> |

Ad hoc proliferation of values for radius is going to create a maintenance headache down the road, so it would be good to think systematically about a refreshed radius scale, and semantic application thereof.

### Radius scale

<img width="821" alt="radius2" src="https://github.com/user-attachments/assets/ca31cb2a-2be0-4875-9ca1-165e46bde0d0">

Members of the design team have been looking into this and formulated the following scale/guidelines:

- `$radius-x-small`: The smallest radius. Applied to contained elements within primitives like inputs or buttons. E.g. the selected item in a `ToggleGroupControl`, or the unit selector in `UnitControl`.
- `$radius-small`: Applied to primitive components like Badges, Buttons, Inputs, Checkboxes.
- `$radius-medium`: Applied to containers with smaller padding, e.g. Menus, Notices, Snackbars, Block toolbar. Also used for focus rings on primitive components.
- `$radius-large`: Applied to containers with larger padding, e.g. Modals, Pages, and the Preview Frame.
- `$radius-full`: For creating lozenges.
- `$radius-round`: For creating circles or ovals. (This variable already exists).

Note: The _values_ assigned to each item in the scale can be decided separately, but we can discuss those here too.

### Implementation

Once we find consensus around the scale and values, a possible implementation plan might look like:

- [x] https://github.com/WordPress/gutenberg/pull/64007
- [x] https://github.com/WordPress/gutenberg/pull/64368
- [x] https://github.com/WordPress/gutenberg/pull/64930
- [ ] Block toolbar / 'Toolbar' variant of dropdown menus
- [x] https://github.com/WordPress/gutenberg/pull/66219

**Comment by ciampo** [comment]
Some feedback points on the naming:

- it may be confusing to know in advance which radius is bigger between "atom" vs "primitive"
- I find that mixing abstract names like "atom" and "primitive" with "container" feels a bit incoherent.

How do other design systems call their radius scale?

Also, should this be a 1:1 of what will be included for radii in the `theme` package?

**Comment by hanneslsm** [comment]
> How do other design systems call their radius scale?

Usually xs, sm, md, lg in various spellings
https://tailwindcss.com/docs/border-radius
https://developer.apple.com/documentation/tvml/border-radius
https://m3.material.io/styles/shape/overview

**Comment by jameskoster** [comment]
Good points. I was thinking there could be two parts;

* The scale itself which should likely use a naming convention like xs, sm, md, lg, as suggested
* Semantic tokens that help clarify which radius to use in a given situation. Names TBD :)

Perhaps the latter is unnecessary?

**Comment by ciampo** [comment]
> xs, sm, md, lg, as suggested

That is definitely clearer to understand. The alternative is to go even more "literal" and use a number-based scale ([example](https://polaris.shopify.com/tokens/border))

> Semantic tokens that help clarify which radius to use in a given situation. Names TBD :)
> Perhaps the latter is unnecessary?

The risk of using semantic names is that, depending on what names are chosen, they can get "out of sync" if the design direction updates. Are there other design systems adopting semantic token names on top of a "neutral" scale?

Maybe a good way to move forward is to start without semantic tokens, get a feel for it, and re-discuss if we feel like semantic tokens would be a nice addition?

**Comment by jameskoster** [comment]
Semantic tokens is something we've been exploring in Figma for colors. But it might not need to be codified, especially for less-used scales like Radius. 

Starting without tokens seems very reasonable.

**Comment by mattrwalker** [comment]
I agree with the Polaris example of naming being clearer but also potentially presenting risk if specific values are ever needed updating. The t-shirt size approach is a great iteration of the naming that feels clearer while also allowing us some flexibility to adjust values in the future too, I'd be in favor of the T-shirt size naming whenever we're ready to create tokens.

We could also ensure that descriptions and usage guidance for each radii are defined and communicated clearly whenever it's referenced to ensure folks know how and when to use the right radius. 

Great breakdown per usual @jameskoster
